### PR TITLE
Test the publishing flow

### PR DESCRIPTION
**Pull request by ghickman**
Now that publishing is just about considered done and we've documented it, this adds an integration test to codify the flow.

While adding this test I discovered the PublishRequestCreate page doesn't handle a report being locked.  I've mirrored the ReportEdit page's flow, showing a locked template in that situation:

### CODE REVIEW

File: tests/integration/test_interactive.py [hunk 1]
```diff
@@ -95,3 +118,110 @@ def test_interactive_submission_success(rf, local_repo, enable_network):
...
+    # request it's published
+    assert analysis.status == "succeeded"
+
+    # user can view their analysis page and the report is rendered there
+    with patch.object(
+        AnalysisRequestCreate, "get_opencodelists_api", new=FakeOpenCodelistsAPI
+    ):
+        response = client.get(project.get_interactive_url())
+        assert response.status_code == 200
+        # TODO: check rendered response has the report details
```
Comment (lucyb) [inline tests/integration/test_interactive.py:155]: This TODO is confusing. Can this be implemented or removed, please?

File: tests/integration/test_interactive.py [hunk 2]
```diff
@@ -95,3 +118,110 @@ def test_interactive_submission_success(rf, local_repo, enable_network):
...
+    with patch.object(PublishRequestCreate, "get_github_api", new=FakeGitHubAPI):
+        response = client.post(analysis.get_publish_url(), follow=True)
+        assert response.status_code == 200
+        assert response.redirect_chain == [(analysis.get_absolute_url(), 302)]
+
+    # check ticket is requested
+    requests = PublishRequest.objects.filter(report=report)
+    assert requests.count() == 1
+
+    # grab the publish request to make some of the tests below a little easier
```
Comment (lucyb) [inline tests/integration/test_interactive.py:168]: YMMV, but I'm not sure this comment is needed because the code is clear enough without it, I think.

**Review by lucyb — APPROVED** [review]

File: tests/integration/test_interactive.py (hunk 1 above)
Comment (ghickman) [reply tests/integration/test_interactive.py:155]: It referenced the commented out line below which I forgot to fix 😂

Comment (ghickman) [reply tests/integration/test_interactive.py]: And exposed a broken part of the test 🎉

**Pull request merged**